pkg/app/gitops.go

```diff
@@ -130,3 +191,107 @@ func gitOpsConfigFromSecretData(idx int64, secretData map[string][]byte) (string
 
 	return provider, publicKey, privateKey, repoURI, nil
 }
+
+func createGitOpsCommit(gitOpsConfig *GitOpsConfig, appSlug string, appName string, newSequence int, archiveDir string, downstreamName string) (string, error) {
+	kotsKinds, err := kotsutil.LoadKotsKindsFromPath(archiveDir)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to load kots kinds")
+	}
+
+	kustomizeVersion := "3.5.4"
+	if kotsKinds.KotsApplication.Spec.KustomizeVersion != "" {
+		kustomizeVersion = kotsKinds.KotsApplication.Spec.KustomizeVersion
+	}
+
+	// we use the kustomize binary here...
+	cmd := exec.Command(fmt.Sprintf("kustomize%s", kustomizeVersion), "build", filepath.Join(archiveDir, "overlays", "downstreams", downstreamName))
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to run kustomize")
+	}
+
+	// using the deploy key, create the commit in a new branch
+	var auth transport.AuthMethod
+	signer, err := ssh.ParsePrivateKey([]byte(gitOpsConfig.PrivateKey))
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse deploy key")
+	}
+	auth = &go_git_ssh.PublicKeys{User: "git", Signer: signer}
+	auth.(*go_git_ssh.PublicKeys).HostKeyCallback = ssh.InsecureIgnoreHostKey()
+
+	workDir, err := ioutil.TempDir("", "kotsadm")
+	if err != nil {
+		return "", errors.Wrap(err, "failed to create temp dir")
+	}
+
+	cloned, err := git.PlainClone(workDir, false, &git.CloneOptions{
```

Comment:
any way we can just do this in memory? https://github.com/src-d/go-git#in-memory-example
long-term goal, I suppose.

If we're not doing this in-memory, we should have a `defer ioutil.RemoveAll(workDir)` before this